ggml/src/ggml-cuda/dmmv.cu

```diff
@@ -416,10 +416,11 @@ static __global__ void dequantize_mul_mat_vec_q6_k(const void * __restrict__ vx,
 
 static __device__ void convert_f16(const void * vx, const int64_t ib, const int iqs, dfloat2 & v){
     const half * x = (const half *) vx;
-
+    // load 2 halfs into register in a single instruction
+    const half2 x_reg = *((half2 *) &(x[ib + iqs]));
     // automatic half -> float type cast if dfloat == float
-    v.x = x[ib + iqs + 0];
-    v.y = x[ib + iqs + 1];
+    v.x = x_reg.x;
+    v.y = x_reg.y;
```

Comment:
This needs to use instructions like `__low2float` in order to work correctly with HIP. Also did you check the PTX code regarding whether or not these two lines are equivalent to assigning `half2` directly?

Reply:
I've now replaced with `__low2float` and `__high2float`. NVCC gives an error if I try an assign x_reg to v directly, but these ops are in-register so won't be on the critical path anyway